app/models/spree/gateway/adyen_credit_card.rb

```diff
@@ -108,6 +124,27 @@ def reference_number_from_order order
       order.user_id.to_s.presence || order.number
     end
 
+    def authorization_3d_request payment, redirect_response_params
+      request = {
```

Comment:
This is also largely duplicating `authorization_request`, could this just merge in the additional keys it needs?